kube/registry_credentials_test.go

```diff
@@ -0,0 +1,123 @@
+package kube
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/SUSE/fissile/testhelpers"
+)
+
+func TestMakeRegistryCredentialsForKube(t *testing.T) {
+	assert := assert.New(t)
+
+	rcreds, err := MakeRegistryCredentials(ExportSettings{})
+
+	if !assert.NoError(err) {
+		return
+	}
+
+	rcredsYAML, err := testhelpers.RenderNode(rcreds, nil)
+	// config - Helm only
+
+	if !assert.NoError(err) {
+		return
+	}
+
+	expectedYAML := `---
+apiVersion: "v1"
+data:
+  .dockercfg: ""
+kind: "Secret"
+metadata:
+  name: "registry-credentials"
+  labels:
+    skiff-role-name: "registry-credentials"
+type: "kubernetes.io/dockercfg"
+`
+	assert.Equal(expectedYAML, string(rcredsYAML))
+}
+
+func TestMakeRegistryCredentialsForHelmWithDefaults(t *testing.T) {
+	assert := assert.New(t)
+
+	rcreds, err := MakeRegistryCredentials(ExportSettings{
+		CreateHelmChart: true,
+	})
+
+	if !assert.NoError(err) {
+		return
+	}
+
+	rcredsYAML, err := testhelpers.RenderNode(rcreds, nil)
+	// config - Helm only
+	// .Values.kube.registry.hostname
+	// .Values.kube.registry.username
+	// .Values.kube.registry.password
+	// json, base64 encoded
+
+	if !assert.NoError(err) {
+		return
+	}
+
+	// Notes: The base64 decodes to
+	// 	{%!q(<nil>):{"username":"","password":"","auth":"JSFzKDxuaWw+KTolIXMoPG5pbD4p"}}
+	// and the auth value decodes to
+	// 	%!s(<nil>):%!s(<nil>)
+
+	expectedYAML := `---
+apiVersion: "v1"
+data:
+  .dockercfg: eyUhcSg8bmlsPik6eyJ1c2VybmFtZSI6IiIsInBhc3N3b3JkIjoiIiwiYXV0aCI6IkpTRnpLRHh1YVd3K0tUb2xJWE1vUEc1cGJENHAifX0=
```

Comment:
We should to base64 encoding in the test, so that it's easier to understand what it's doing.

Reply:
Hm. Will look into it.

Reply:
Commit 1e1528333fcb8ec0e8b611b72b1be9127c7d8350.